File: app/src/main/java/de/mchllngr/quickopen/util/SplashScreenHelper.java
package de.mchllngr.quickero.util;

import android.app.Activity;
import android.app.Application;
import android.content.pm.ActivityInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;

import androidx.annotation.NonNull;
import de.mchllngr.quickero.R;
import timber.log.Timber;

/**
 * http://blog.davidmedenjak.com/android/2017/09/02/splash-screens.html
 */
public class SplashScreenHelper implements Application.ActivityLifecycleCallbacks {

    private static final String KEY_THEME = "theme";

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
        // apply the actual theme given by meta-data
        try {
            ActivityInfo activityInfo = activity.getPackageManager().getActivityInfo(activity.getComponentName(), PackageManager.GET_META_DATA);

            int theme = R.style.AppTheme;
            if (activityInfo.metaData != null) {
                theme = activityInfo.metaData.getInt(KEY_THEME, R.style.AppTheme);
            }

            activity.setTheme(theme);
        } catch (PackageManager.NameNotFoundException e) {
            Timber.e(e);
        }
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) { /* empty */ }

    @Override
    public void onActivityResumed(@NonNull Activity activity) { /* empty */ }

    @Override
    public void onActivityPaused(@NonNull Activity activity) { /* empty */ }

    @Override
    public void onActivityStopped(@NonNull Activity activity) { /* empty */ }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle bundle) { /* empty */ }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) { /* empty */ }
}

File: app/src/main/java/de/mchllngr/quickero/util/splash/SplashScreenHelper.kt
package de.mchllngr.quickero.util.splash

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.content.pm.PackageManager
import android.os.Bundle
import de.mchllngr.quickero.R
import timber.log.Timber

/* http://blog.davidmedenjak.com/android/2017/09/02/splash-screens.html */
class SplashScreenHelper : ActivityLifecycleCallbacks {

    override fun onActivityCreated(
        activity: Activity,
        savedInstanceState: Bundle?
    ) {
        // apply the actual theme given by meta-data
        try {
            val activityInfo = activity.packageManager.getActivityInfo(activity.componentName, PackageManager.GET_META_DATA)
            val theme = when {
                activityInfo.metaData != null -> activityInfo.metaData.getInt(KEY_THEME, R.style.AppTheme)
                else -> R.style.AppTheme
            }
            activity.setTheme(theme)
        } catch (e: PackageManager.NameNotFoundException) {
            Timber.e(e)
        }
    }

    override fun onActivityStarted(activity: Activity) = Unit

    override fun onActivityResumed(activity: Activity) = Unit

    override fun onActivityPaused(activity: Activity) = Unit

    override fun onActivityStopped(activity: Activity) = Unit

    override fun onActivitySaveInstanceState(
        activity: Activity,
        bundle: Bundle
    ) = Unit

    override fun onActivityDestroyed(activity: Activity) = Unit

    companion object {

        private const val KEY_THEME = "theme"
    }
}
